src/pages/ReportAddApproverPage.tsx

```diff
@@ -0,0 +1,205 @@
+    const [selectedApproverEmail, setSelectedApproverEmail] = useState<string | undefined>(undefined);
+    const [personalDetails] = useOnyx(ONYXKEYS.PERSONAL_DETAILS_LIST, {canBeMissing: false});
+    const [countryCode] = useOnyx(ONYXKEYS.COUNTRY_CODE, {canBeMissing: false});
+
+    const currentUserDetails = useCurrentUserPersonalDetails();
+
+    const [allApprovers, setAllApprovers] = useState<SelectionListApprover[]>([]);
+    const shouldShowTextInput = allApprovers?.length >= CONST.STANDARD_LIST_ITEM_LIMIT;
+
+    const employeeList = policy?.employeeList;
+    const sections = useMemo(() => {
+        const approvers: SelectionListApprover[] = [];
+
+        if (employeeList) {
+            const policyMemberEmailsToAccountIDs = getMemberAccountIDsForWorkspace(employeeList);
+            const availableApprovers = Object.values(employeeList)
+                .map((employee): SelectionListApprover | null => {
+                    const isAdmin = employee?.role === CONST.REPORT.ROLE.ADMIN;
+                    const email = employee.email;
+
+                    if (!email) {
+                        return null;
+                    }
+                    const accountID = Number(policyMemberEmailsToAccountIDs[email] ?? CONST.DEFAULT_NUMBER_ID);
+
+                    if (report.managerID === accountID) {
+                        return null;
+                    }
+
+                    const {avatar} = personalDetails?.[accountID] ?? {};
+                    const displayName = getDisplayNameForParticipant({accountID, personalDetailsData: personalDetails});
+                    return {
+                        text: displayName,
+                        alternateText: email,
+                        keyForList: email,
+                        isSelected: selectedApproverEmail === email,
+                        login: email,
+                        value: accountID.toString(),
+                        icons: [{source: avatar ?? FallbackAvatar, type: CONST.ICON_TYPE_AVATAR, name: displayName, id: accountID}],
+                        rightElement: isAdmin ? <Badge text={translate('common.admin')} /> : undefined,
+                    };
+                })
+                .filter((approver): approver is SelectionListApprover => !!approver);
+
+            approvers.push(...availableApprovers);
+            // eslint-disable-next-line react-compiler/react-compiler
+            setAllApprovers(approvers);
+        }
+
+        const filteredApprovers =
+            debouncedSearchTerm !== ''
+                ? tokenizedSearch(approvers, getSearchValueForPhoneOrEmail(debouncedSearchTerm, countryCode), (option) => [option.text ?? '', option.login ?? ''])
+                : approvers;
+
+        const data = sortAlphabetically(filteredApprovers, 'text', localeCompare);
+        return [
+            {
+                title: undefined,
+                data,
+                shouldShow: true,
+            },
+        ];
```

Comment:
This looks like copy pasta from the `WorkspaceWorkflowsApprovalsApproverPage`. I have some kind of concern that if the code changes in one place we will neglect to update it in the other place. Is there some opportunity to refactor here and avoid any duplication? I haven't looked too deeply yet so mostly looking to hear your thoughts.